app/scripts/proactive/view/dom.js

```diff
@@ -326,6 +326,170 @@ define(
+            $("#ul-calling").append(pinOpen);
+            var menuElement = $('#ul-calling').parent();
+            if (!menuElement.hasClass('dropdown')) {
+                $("#li-pin-calling").html('<img src="images/icon-unpin.png"> Unpin');
+            }
+            //pinUnpin function pin/unpin the menu
+            $("#li-pin-calling").click(function(event) {
+                pinUnpin($("#ul-calling"), $("#li-pin-calling"));
+            });
+
+            event.preventDefault();
+            var studioApp = require('StudioApp');
+            var tasks = studioApp.models.jobModel.tasks;
+            var TasksArray = [];
+            for (var i = 0; i < tasks.length; i++) {
+                TasksArray[i] = tasks[i];
+            }
+            //isCallingObjects becomes true when there is at least one called catalog object
+            var isCallingObjects = false;
+            for (var i = 0; i < TasksArray.length; i++) {
+                var task = TasksArray[i];
+                if (task.has('Variables')) {
+                    var variables = task.get('Variables');
+                    var taskName = "previousTask";
+                    for (var j = 0; j < variables.length; j++) {
+                        var variable = variables[j];
+                        if (variable.Model && variable.Model.includes("PA:CATALOG_OBJECT")) {
+                            isCallingObjects = true;
+                            var tName = task.get('Task Name');
+                            //we add the task name only the first time, else we print empty string
+                            if (tName != taskName && taskName != "") {
+                                taskName = tName;
+                            } else {
+                                taskName = "";
+                            }
+                            var variableValue = variable.Value;
+                            if (variableValue) {
+                                var taskViewModel = studioApp.views.workflowView.taskViews[0];
+                                var calledObjectDetails = taskViewModel.getCalledObjectDetails(variableValue)
+                                var objectKind = taskViewModel.getObjectKind(calledObjectDetails["bucketName"], calledObjectDetails["objectName"])
+                                if (objectKind == 'Workflow/standard' || objectKind == 'Workflow/psa') {
+                                    var menuItemCalling = $('<li class="sub-menu draggable ui-draggable job-element"><a style="display: table-cell; padding:3px 10px;min-width: 180px;">' + taskName + '</a><a style="display: table-cell; padding:3px 10px"> ' + variableValue + '</a><a href="/studio/#workflowcatalog/' + calledObjectDetails["bucketName"] + '/workflow/' + calledObjectDetails["objectName"] + '" target="_blank" style="color:#337ab7; display: table-cell;"><i class="glyphicon glyphicon-eye-open"></i></a></li>');
```

Comment:
some html strings are repeated many times in this code, you could introduce local variables to avoid repetition